fs-storage/src/file_storage.rs

```diff
@@ -219,4 +226,28 @@ mod tests {
         }
         assert_eq!(storage_path.exists(), false);
     }
+
+    #[test]
+    fn test_file_storage_is_storage_updated() {
```

Comment:
Tracing the error back. seems like the it is because the nanoseconds don't match. `SystemTime` stores the time in nanosecond precision
```sh
running 1 test
Self timestamp: SystemTime { tv_sec: 1713553086, tv_nsec: 320082935 }
File timestamp: SystemTime { tv_sec: 1713553086, tv_nsec: 320194727 }
thread 'file_storage::tests::test_file_storage_is_storage_updated' panicked at fs-storage/src/file_storage.rs:243:9:
assertion `left == right` failed
  left: true
 right: false
```

I don't think we need nanosecond precision. Maybe we can just compare seconds
```rust
fn is_storage_updated(&self) -> Result<bool> {
	let file_timestamp = fs::metadata(&self.path)?.modified()?;
	let file_time_secs = file_timestamp
		.duration_since(SystemTime::UNIX_EPOCH)
		.unwrap() // SAFETY: We can safely unwrap because the timestamp is always after the epoch
		.as_secs();
	let self_time_secs = self
		.timestamp
		.duration_since(SystemTime::UNIX_EPOCH)
		.unwrap()
		.as_secs();
	Ok(file_time_secs > self_time_secs)
}
```
But you might have a better idea.

Reply:
@tareknaser probably we need more granular timestamps. It's ok to merge it like this, but let's think about such scenario:
1. We scan folder at the moment `HH:MM:ss:xxx`, where `xxx` is milliseconds part.
2. A file `cat.jpg` is modified at the moment `HH:MM:ss:yyy`.

If we truncate `xxx` and `yyy` then on next scan we'll skip `cat.jpg` and we miss the update.

As a workaround, it seems possible rounding `HH:MM:ss:xxx` always to the lowest side, and `HH:MM:ss:yyy` always to the highest side. This would result in repeated updates for the files which were updated in the same second when scan happened, but it's better than to lose an update.

Also, I think we have discussed somewhere the smallest division of time that makes sense for file systems... Nanoseconds are too low, but what about milliseconds?

@twitu